googletest/MockDevice.hpp

```diff
@@ -0,0 +1,1247 @@
+        return;
+    }
+    
+    virtual HRESULT STDMETHODCALLTYPE CreateQueryHeap( 
+        _In_  const D3D12_QUERY_HEAP_DESC *pDesc,
+        REFIID riid,
+        _COM_Outptr_opt_  void **ppvHeap) override 
+    {
+        return S_OK;    
+    }
+    
+    virtual HRESULT STDMETHODCALLTYPE SetStablePowerState( 
+        BOOL Enable) override 
+    {
+        return S_OK;
+    }
+    
+    virtual HRESULT STDMETHODCALLTYPE CreateCommandSignature( 
+        _In_  const D3D12_COMMAND_SIGNATURE_DESC *pDesc,
+        _In_opt_  ID3D12RootSignature *pRootSignature,
+        REFIID riid,
+        _COM_Outptr_opt_  void **ppvCommandSignature) override
+    {
+        return S_OK;
+    }
+    
+    virtual void STDMETHODCALLTYPE GetResourceTiling( 
+        _In_  ID3D12Resource *pTiledResource,
+        _Out_opt_  UINT *pNumTilesForEntireResource,
+        _Out_opt_  D3D12_PACKED_MIP_INFO *pPackedMipDesc,
+        _Out_opt_  D3D12_TILE_SHAPE *pStandardTileShapeForNonPackedMips,
+        _Inout_opt_  UINT *pNumSubresourceTilings,
+        _In_  UINT FirstSubresourceTilingToGet,
+        _Out_writes_(*pNumSubresourceTilings)  D3D12_SUBRESOURCE_TILING *pSubresourceTilingsForNonPackedMips) override 
+    {
+        return;
+    }
+    
+    virtual LUID STDMETHODCALLTYPE GetAdapterLuid( void) override 
+    {
+        LUID mockLuid = {};
+        return mockLuid;
+    }
+
+public: // ID3D12Object
+    virtual HRESULT STDMETHODCALLTYPE GetPrivateData( 
+        _In_  REFGUID guid,
+        _Inout_  UINT *pDataSize,
+        _Out_writes_bytes_opt_( *pDataSize )  void *pData) override
+    {
+        return S_OK;
+    }
+    
+    virtual HRESULT STDMETHODCALLTYPE SetPrivateData( 
+        _In_  REFGUID guid,
+        _In_  UINT DataSize,
+        _In_reads_bytes_opt_( DataSize )  const void *pData) override
+    {
+        return S_OK;
+    }
+    
+    virtual HRESULT STDMETHODCALLTYPE SetPrivateDataInterface( 
+        _In_  REFGUID guid,
+        _In_opt_  const IUnknown *pData) override 
+    {
+        return S_OK;
+    }
+    
+    virtual HRESULT STDMETHODCALLTYPE SetName( 
+        _In_z_  LPCWSTR Name) override 
+    {
+        return S_OK;
+    }
+
+public: // IUnknown
+    virtual HRESULT STDMETHODCALLTYPE QueryInterface(
+        /* [in] */ REFIID riid,
+        /* [iid_is][out] */ _COM_Outptr_ void __RPC_FAR *__RPC_FAR *ppvObject) override
+    {
+        *ppvObject = this;
+        return S_OK;
+    }
+
+    virtual ULONG STDMETHODCALLTYPE AddRef() override 
+    {
+        // Casual implementation. No actual actions
+        return 0;
+    }
+
+    virtual ULONG STDMETHODCALLTYPE Release() override 
+    {
+        return 0;
+    }
+
+
+    // Major function we need to work with
+    virtual HRESULT STDMETHODCALLTYPE CheckFeatureSupport( 
+        D3D12_FEATURE Feature,
+        _Inout_updates_bytes_(FeatureSupportDataSize)  void *pFeatureSupportData,
+        UINT FeatureSupportDataSize
+    ) override 
+    {
+        switch( Feature )
```

Comment:
I removed features that are not present in this repo's `d3d12.h` and removed all internal query functions, but I want to double check I'm not leaking any code that's not prepared for release.